src/foam/u2/texteditor/Popup.js

```diff
@@ -0,0 +1,46 @@
+foam.CLASS({
+  package: 'foam.u2.texteditor',
+  name: 'Popup',
+  extends: 'foam.u2.View',
+
+  properties: [
+    {
+      name: 'button'
+    },
+    {
+      name: 'view'
+    },
+    {
+      class: 'Boolean',
+      name: 'hidden',
+      value: true
+    }
+  ],
+  methods: [
+    function initE() {
+      var self = this;
+      self.SUPER();
+      
+      self
+          .start()
+            .add(self.button$)
+            .on('click', function(e1) {
+              self.hidden = ! self.hidden;
+              if ( ! self.hidden ) {
+                self.document.addEventListener('click', function(e2) {
+                  if ( e1.target === e2.target ) return;
+                  if ( ! self.hidden ) self.hidden = true;
+                  self.document.removeEventListener('click', arguments.callee);
+                });
+              }
+            })
+          .end()
+          .start('div').addClass('toolbar-menu')
+            .style({
```

Comment:
I recently learned of the "hide" method in u2 that you can use here instead.
You should be able to replace this call to `style` with a call to `hide(self.hidden$)`